kerchunk/df.py

```diff
@@ -123,242 +128,102 @@ def refs_to_dataframe(
-                key = f"{field}/{metakey}"
-                _write_json(os.path.join(field_path, metakey), refs[key])
-            for i, ind in enumerate(np.ndindex(tuple(chunk_sizes.astype(int)))):
-                chunk_id = ".".join([str(ix) for ix in ind])
-                key = f"{field}/{chunk_id}"
-                # Make note if expected number of chunks differs from actual
-                # number found in references
-                if key in refs:
-                    data = refs[key]
-                    if isinstance(data, list):
-                        paths[i] = data[0]
-                        offsets[i] = data[1]
-                        sizes[i] = data[2]
-                    else:
-                        raws[i] = _proc_raw(data)
+        for i, ind in enumerate(np.ndindex(tuple(chunk_sizes.astype(int)))):
+            chunk_id = ".".join([str(ix) for ix in ind])
+            key = f"{field}/{chunk_id}"
+            # Make note if expected number of chunks differs from actual
+            # number found in references
+            if key in refs:
+                data = refs[key]
+                if isinstance(data, list):
+                    paths[i] = data[0]
+                    offsets[i] = data[1]
+                    sizes[i] = data[2]
                 else:
-                    nmissing += 1
-
-            if nmissing:
-                print(
-                    f"Warning: Chunks missing for field {field}. "
-                    f"Expected: {nchunks}, Found: {nchunks - nmissing}"
-                )
-            # The convention for parquet files is
-            # <url>/<field_name>/refs.parq
-            out_path = os.path.join(field_path, "refs.parq")
-            df = pd.DataFrame(
-                dict(path=paths, offset=offsets, size=sizes, raw=raws), copy=False
-            )
-            # Engine specific kwarg conventions. Set stats to false since
-            # those are currently unneeded.
-            if engine == "pyarrow":
-                # TODO: perhaps set some other defaults to pyarrow engine for
-                # optimization?
-                kwargs.update(write_statistics=False, row_group_size=row_group_size)
+                    raws[i] = _proc_raw(data)
             else:
-                kwargs.update(
-                    row_group_offsets=row_group_size,
-                    object_encoding=dict(raw="bytes", path="utf8"),
-                    has_nulls=["path", "raw"],
-                    stats=False,
-                )
-            df.to_parquet(
-                out_path,
-                engine=engine,
-                storage_options=storage_options,
-                compression=compression,
-                index=False,
-                **kwargs,
+                nmissing += 1
+
+        if nmissing:
+            # comment: missing keys are fine, so long as they are not a large fraction.
+            #  Does referenceFS successfully give FileNotFound for them?
```

Comment:
I added this check since I was initially testing this with another reference set where the expected number of chunks didn't match the actual number of files available. The `LazyReferenceMapper` needs row group sizes to be consistent so missing references end up being set to nan when the arrays are loaded. Do you think raising the FileNotFoundError would be a better default behavior for missing refs in ReferenceFS?